File: TmpFiles/math.py
import numpy as np

def a(n):
    if n % 2 == 0: return - 1 / np.sqrt(n + 1)
    else: return 1 / np.sqrt(n + 1)

N = 10**9
Pn = 1
for i in range(1, N):
    Pn *= (1 + a(i))
print(Pn)

File: TmpFiles/math.cpp
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

double a(int n) {
    if (n % 2 == 0) return -1/sqrt(n+1);
    else return 1/sqrt(n+1);
}

int main(int argc, char *argv[]) {
    double P = 1;
    int N = stoi(argv[1]);
    for (int n = 1; n < N; n++) {
        P *= a(n);
    }
    cout << P << endl;
    return 0;
}
